src/briefcase/integrations/java.py

```diff
@@ -284,3 +288,24 @@ def upgrade(self):
 
         self.uninstall()
         self.install()
+
+    @classmethod
+    def verify_rosetta(cls, tools):
+        try:
+            tools.subprocess.check_output(
+                ["arch", "-x86_64", "true"], stderr=subprocess.STDOUT
```

Comment:
Does this reliably test CPU architecture? `true` is a shell builtin, so I wouldn't have thought it will trigger the CPU check. Running `/bin/bash -version` might be a better option - with the added benefit that as debug, it leaves a paper trail:
```
(venv3.10) rkm@eunectes testbed % arch -arm64 /bin/bash -version 
GNU bash, version 3.2.57(1)-release (arm64-apple-darwin21)
Copyright (C) 2007 Free Software Foundation, Inc.
(venv3.10) rkm@eunectes testbed % arch -x86_64 /bin/bash -version 
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin21)
Copyright (C) 2007 Free Software Foundation, Inc.
```

Reply:
Most shell builtins that can exist as separate executables, actually do: 
```
(bw-311) msmith@mbp briefcase % where echo
echo: shell built-in command
/bin/echo
(bw-311) msmith@mbp briefcase % where true
true: shell built-in command
/usr/bin/true
```

Since `arch` doesn't run a shell, it'll try to run /usr/bin/true. And with Rosetta uninstalled, the check fails correctly:
```
(bw-311) msmith@mbp briefcase % arch -x86_64 true
arch: posix_spawnp: true: Bad CPU type in executable
(bw-311) msmith@mbp briefcase % echo $?
1
```